account_invoice_report_by_partner/models/account.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Angel Moya <angel.moya@domatix.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def invoice_print(self):
+        self.ensure_one()
+        self.sent = True
```

Comment:
I think it's really better to call the super method for that. After you can change the result by the new action